tls/s2n_config.c

```diff
@@ -1138,3 +1139,45 @@ int s2n_config_set_cert_validation_cb(struct s2n_config *config, s2n_cert_valida
 
     return S2N_SUCCESS;
 }
+
+int s2n_config_get_supported_groups(struct s2n_config *config, uint16_t *groups, uint16_t groups_count_max,
+        uint16_t *groups_count_out)
+{
+    POSIX_ENSURE_REF(groups_count_out);
+    *groups_count_out = 0;
+    POSIX_ENSURE_REF(config);
+    POSIX_ENSURE_REF(groups);
+
+    const struct s2n_security_policy *security_policy = config->security_policy;
+    POSIX_ENSURE_REF(security_policy);
+    const struct s2n_kem_preferences *kem_preferences = security_policy->kem_preferences;
+    POSIX_ENSURE_REF(kem_preferences);
+    const struct s2n_ecc_preferences *ecc_preferences = security_policy->ecc_preferences;
+    POSIX_ENSURE_REF(ecc_preferences);
+
+    uint16_t groups_count = 0;
+    for (uint8_t i = 0; i < kem_preferences->tls13_kem_group_count; i++) {
```

Comment:
That's because only the TLS 1.3 kem groups are negotiated in the supported groups extension. I think in TLS 1.2 the client indicates support for PQ kems with the cipher suite and then negotiates them in the key exchange message.

Reply:
Hmm yeah I see what you're saying. But I'm not sure I agree that we'd want to include the earlier groups, even if there wasn't a motivation to link this API with the client hello supported groups getter. 

I guess I don't really consider the PQ kems sent in TLS 1.2 to be "supported groups", but let me know if you disagree. The PQ KEM extension seems similar to the supported groups extension, except that it [only lists PQ kems](https://datatracker.ietf.org/doc/html/draft-campagna-tls-bike-sike-hybrid#section-5.1.6). The values corresponding to each of the PQ kems is unique to this extension. This means that the values could (and seem to) conflict with the [IANA values in the supported groups extension](https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-8). This would make it confusing to users as to which groups is actually being returned from this API.

If this still seems weird though, would it help if we split up this API to return more specific values from the security policy? So for example this API could just be `s2n_config_get_ecc_curvesI()`, and we could then later add `s2n_config_get_tls12_pq_kems()` and `s2n_config_get_tls13_pq_kems()` if this is needed.

Reply:
Oh wow, what a pain, the PQ iana values haven't even been standardized yet 😭. 
Honestly I don't think it's better to split this up into (what would eventually become) 3 APIs. But we definitely need a note in the API documentation about the limitations of this API.